### ISSUE TITLE
Timeouts in tests in CI

### ISSUE THREAD

**Issue by volkm** [moves-rwth/storm#514]
Recently, the Build Test in the CI have been failing. I thought it was due to the way we obtain the z3 version, but the [improved version parsing](https://github.com/moves-rwth/storm/pull/513) still fails in the test phase.

I believe that the address sanitizer triggers the issue. Disabling the compile flag `STORM_COMPILE_WITH_ALL_SANITIZERS` seems to work, see [here](https://github.com/volkm/storm/actions/runs/8323643800). I have no idea yet what exactly triggers the issue though.

**Comment by tquatmann** [comment]
Could it be that the executable gets stuck after a sanitizer fails? There is the `-fsanitize-recover=address` flag which might be relevant here.
Can we run the tests with output? Maybe this shows something useful.

**Comment by volkm** [comment]
It becomes even weirder, because now the build tests are again running perfectly.
I compared log files and the only notable difference I found was that the underlying Ubuntu image in the CI changed (it is updated weekly). It seems that all builds with Github runner version 20240310.1.0 failed for a week and the previous and later versions work fine. I also found some [related issue](https://github.com/actions/runner-images/issues/9491). I find it interesting that the issue propagated through the Docker container, because we are running our own Docker on the provided Ubuntu image.
